sizzlingstats/SSTeamData.cpp

```diff
@@ -0,0 +1,79 @@
+
+#include "SSTeamData.h"
+
+void TeamScoreData::Reset()
+{
+	m_AverageMedicDistance.Reset();
+}
+
+CTeamDataManager::CTeamDataManager()
+{
+
+}
+
+CTeamDataManager::~CTeamDataManager()
+{
+}
+
+void CTeamDataManager::Reset()
+{
+	TeamScoreData *tm_data = nullptr;
+	
+	if((tm_data = GetTeamStats(TFTEAM_RED)) != nullptr)
+	{
+		tm_data->Reset();
+	}
+
+	if((tm_data = GetTeamStats(TFTEAM_BLUE)) != nullptr)
+	{
+		tm_data->Reset();
+	}
+
+}
+
+void CTeamDataManager::ResetTeamStatsData()
+{
+}
+
+void CTeamDataManager::AddStatSample( int team, TeamStat stat, float sample_point )
+{
+	if(!IsTFTeam((tfteam)team))
+		return;
+
+	if(stat == TeamStat::MedicCohesion)
+	{
+		TeamScoreData *team_data = GetTeamStats((tfteam)team);
+		if(!team_data)
+			return;
+
+		team_data->m_AverageMedicDistance(sample_point);
+	}
+}
+
+double CTeamDataManager::GetStat( int team, TeamStat stat )
+{
+	if(!IsTFTeam((tfteam)team))
+		return 0.0;
+
+	if(stat == TeamStat::MedicCohesion)
+	{
+		TeamScoreData *team_data = GetTeamStats((tfteam)team);
+		if(!team_data)
+			return 0.0;
+		return team_data->m_AverageMedicDistance;
+	}
+
+	return 0.0;
+}
+
+
+TeamScoreData *CTeamDataManager::GetTeamStats( tfteam team)
```

Comment:
Since you have the team in enum type, you can assume the value will be in range. I'd assume it would be ok to just index the array directly with the team variable.